opentelemetry/proto/common/v1/common.proto

```diff
@@ -21,34 +21,41 @@ option java_package = "io.opentelemetry.proto.common.v1";
 option java_outer_classname = "CommonProto";
 option go_package = "github.com/open-telemetry/opentelemetry-proto/gen/go/common/v1";
 
-// AttributeKeyValue is a key-value pair that is used to store Span attributes, Link
-// attributes, etc.
-message AttributeKeyValue {
-  // ValueType is the enumeration of possible types that value can have.
-  enum ValueType {
-    STRING  = 0;
-    INT     = 1;
-    DOUBLE  = 2;
-    BOOL    = 3;
-  };
-
-  // key part of the key-value pair.
-  string key = 1;
+// AnyValue is used to represent any type of attribute value. AnyValue may contain a
+// primitive value such as a string or integer or it may contain an arbitrary nested
+// object containing arrays, key-value lists and primitives.
+message AnyValue {
+  // The value is one of the listed fields. It is valid for all values to be unspecified
+  // in which case this AnyValue is considered to be "null".
+  oneof value {
+    string string_value = 1;
+    bool bool_value = 2;
+    int64 int_value = 3;
+    double double_value = 4;
+    ArrayValue array_value = 5;
+    KeyValueList kvlist_value = 6;
+  }
+}
 
-  // type of the value.
-  ValueType type = 2;
+message ArrayValue {
+  // Array of values. The array may be empty (contain 0 elements).
+  repeated AnyValue values = 1;
+}
 
-  // Only one of the following fields is supposed to contain data (determined by `type` field).
-  // This is deliberately not using Protobuf `oneof` for performance reasons (verified by benchmarks).
+message KeyValueList {
+  // List of key-value pairs. The list may be empty (contain 0 elements).
+  repeated KeyValue values = 1;
```

Comment:
Is this real list or a map represented as a list? Please comment.

Reply:
From protocol's perspective this is a list of elements. See also trace.proto Span.attributes where we describe it as "a collection of key/value pairs" and avoid calling it a map.

The proto here avoids taking a stance and specifying whether it is a map or multimap (both are possible for a key-value pair list). I have a slight preference to keep it this way to be aligned with how Span.attributes are defined.

I updated the comment here to match what we have for `Span.attributes`.